openedx/features/wikimedia_features/messenger/api/v0/serializers.py

```diff
@@ -0,0 +1,127 @@
+    message = serializers.CharField()
+
+    def validate_receivers(self, receivers):
+        if not receivers:
+            raise serializers.ValidationError('receiver list can not empty.')
+        return receivers
+
+    def bulk_create(self, request=None):
+        created_messages = []
+        if request:
+            receivers = self.validated_data.get('receivers')
+            message = self.validated_data.get('message')
+            with transaction.atomic():
+                for user in receivers:
+                    created_messages.append(
+                        Message.objects.create(sender=request.user, receiver=user, message=message)
+                    )
+        return created_messages
+
+
+class InboxSerializer(serializers.ModelSerializer):
+    with_user = serializers.SerializerMethodField()
+    request = None
+
+    class Meta:
+        model = Inbox
+        fields = ('id', 'with_user', 'last_message', 'unread_count')
+
+    def __init__(self, *args, **kwargs):
+        super(InboxSerializer, self).__init__(*args, **kwargs)
+        self.request = self.context.get('request')
+
+    def _format_last_message_representation(self, instance):
+        # if message length is greater then 20 -> we just want to show first 20 chars with three dots (...)
+        if len(instance.last_message.message) > 20:
+            return "{}...".format(instance.last_message.message[:20])
+        else:
+            return "{}".format(instance.last_message.message)
+
+    def get_with_user(self, obj):
+        if self.request:
+            if obj.last_message.sender != self.request.user:
+                return obj.last_message.sender.username
+            return obj.last_message.receiver.username
+        raise serializers.ValidationError(
+            _('Invalid request - request object not found.')
+        )
+
+    def to_representation(self, instance):
+        response = super().to_representation(instance)
+        with_user = User.objects.get(username=response.get('with_user'))
+        response['with_user_img'] = get_profile_image_urls_for_user(with_user, self.request).get('medium')
+        response['last_message'] = self._format_last_message_representation(instance)
+
+        # if last message is send by login-user then unread count will be 0
+        if self.request and instance.last_message.sender == self.request.user:
+            response['unread_count'] = 0
+
+        return response
+
+
+class ConversationAccessSerializer(serializers.Serializer):
+    user1 = serializers.CharField(max_length=250, write_only=True)
+    user2 = serializers.CharField(max_length=250, write_only=True)
+
+    class Meta:
+        fields = ('user1', 'user2')
+
+    def validate_user1(self, user1):
+        return validate_username(user1)
+
+    def validate_user2(self, user2):
+        return validate_username(user2)
+
+
+class ConversationSerializer(serializers.ModelSerializer):
```

Comment:
this serializer has common fields and methods as `ConversationAccessSerializer`, can we use inheritance here?